src/app/components/shared/datetime-formats/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,217 @@
+const testCases: TestCase[] = [
+  {
+    name: "Luxon DateTime outside test runners timezone",
+    // create a Luxon DateTime object in UTC+00:00
+    // we should see this date be localized to UTC+08:00 (the test runners timezone)
+    // we do not call setZone() here like the other tests because we want to create the DateTime object in UTC+00:00
+    // so that we can see it localized to UTC+08:00
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
+    expectedText: "2020-01-01 20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+  },
+  {
+    name: "Luxon DateTime in test runners timezone",
+    // by setting the timezone to Australia/Perth, we should see that the time value is not changed
+    // because the Luxon DateTime object is already in the same timezone as the test runner
+    // by using { keepLocalTime: true }, we ensure that the time is not changed when we change the timezone
+    // without using { keepLocalTime: true }, the DateTime object would be implicitly created as 20:10:11 +08:00
+    value: DateTime.fromISO("2020-01-01T12:10:11.321Z").setZone(
+      "Australia/Perth",
+      { keepLocalTime: true }
+    ),
+    expectedText: "2020-01-01 12:10:11",
+    expectedTooltip: "2020-01-01 12:10:11.321 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T12:10:11.321+08:00",
+  },
+  {
+    // most of the tests create the DateTime objects in UTC+00:00
+    // for this test, we create th DateTime object in UTC+02:00
+    // we should therefore see the hours increase by 6
+    name: "Luxon DateTime with offset (no timezone)",
+    // by using +02:00, we know the offset, but not the timezone
+    // however, because this value will be localized to the users local timezone, we should see the timezone emitted
+    // in the users local timezone
+    value: DateTime.fromISO("2020-01-01T12:10:11.123+02:00"),
+    expectedText: "2020-01-01 18:10:11",
+    // notice that the tooltips are still in the test runners mock timezone
+    // this is because the emitted date/time has been localized to the test runners timezone
+    expectedTooltip: "2020-01-01 18:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T18:10:11.123+08:00",
+  },
+  {
+    name: "Luxon DateTime with dateOnly",
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
+    expectedText: "2020-01-01",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+    date: true,
+  },
+  {
+    name: "Luxon DateTime with timeOnly",
+    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
+    expectedText: "20:10:11",
+    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
+    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
+    time: true,
+  },
+  {
+    name: "Luxon DateTime with a negative offset (no timezone)",
```

Comment:
```suggestion
    name: "DateTime with a negative offset",
```
Also move this up to be next to the related test